assets/src/features/room-page/utils/usePinning.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+
+export type PinningApi = {
+  pinnedTileIds: string[];
+  pin: (tileId: string) => void;
+  unpin: (tileId: string) => void;
+  pinIfNotAlreadyPinned: (tileId: string) => void;
+};
+
+const usePinning = (): PinningApi => {
```

Comment:
As it is a custom hook, I would rather move this file to `features/room-page/hooks`